odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -1176,6 +1176,27 @@ class SelectionBaseNullImplicit(models.Model):
     ])
 
 
+class SelectionRelated(models.Model):
+    _name = 'test_new_api.model_selection_related'
+    _description = "Model with a related selection field"
+
+    res_id = fields.Many2one(
+        comodel_name='test_new_api.model_selection_base',
+        required=True,
+    )
+    related_selection = fields.Selection(
+        related='res_id.my_selection',
+    )
+
+
+class SelectionRelatedUpdatable(models.Model):
+    _name = 'test_new_api.model_selection_related_updatable'
+    _inherit = 'test_new_api.model_selection_related'
+    _description = "Model with an updatable related selection field"
+
+    related_selection = fields.Selection(readonly=False)
```

Comment:
I had to read this model three times before realizing that it is simply a copy of the model above with `readonly=False` added in the field definition.  Please make a plain stupid copy of the model above.